buildSrc/src/main/kotlin/dokka-conventions.gradle.kts

```diff
@@ -18,62 +11,23 @@ dependencies {
     dokkaPlugin(provider { extens.getByType<VersionCatalogsExtension>().named("libs").findLibrary("dokka.pathsaver").get().get() })
 }
 
-tasks.withType<DokkaTaskPartial>().named("dokkaHtmlPartial") {
-    outputDirectory.set(file("build/dokka"))
-
-
-    pluginsMapConfiguration.put("org.jetbrains.dokka.base.DokkaBase", """{ "templatesDir": "${rootDir.resolve("dokka-templates").canonicalPath.replace('\\', '/')}" }""")
-
-    dokkaSourceSets {
-        configureEach {
-            includes.from(rootDir.resolve("dokka/moduledoc.md").path)
-
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization(\$|\\.).*")
-                reportUndocumented.set(true)
-                skipDeprecated.set(true)
-            }
-
-            // Internal API
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.internal(\$|\\.).*")
-                suppress.set(true)
-            }
-
-            // Internal JSON API
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.json.internal(\$|\\.).*")
-                suppress.set(true)
-                reportUndocumented.set(false)
-            }
-
-            // Workaround for typealias
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.protobuf.internal(\$|\\.).*")
-                suppress.set(true)
-                reportUndocumented.set(false)
-            }
-
-            // Deprecated migrations
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.protobuf(\$|\\.).*")
-                reportUndocumented.set(true)
-                skipDeprecated.set(true)
-            }
+dokka {
+    pluginsConfiguration.html {
+        templatesDir = rootDir.resolve("dokka-templates")
+    }
 
-            // Deprecated migrations
-            perPackageOption {
-                matchingRegex.set("org\\.jetbrains\\.kotlinx\\.serialization\\.config(\$|\\.).*")
-                reportUndocumented.set(false)
-                skipDeprecated.set(true)
-            }
+    dokkaSourceSets.configureEach {
+        includes.from(rootDir.resolve("dokka/moduledoc.md").path)
+        reportUndocumented = true
 
-            sourceLink {
-                localDirectory.set(rootDir)
+        perPackageOption {
```

Comment:
Previously, there were multiple `perPackageOption` - most of them are obsolete or separate configurations for `internal` package. So, obsolete were removed and `internal` are handled by one regex now

Reply:
Are `@Deprecated` declarations skipped now by default?